src/caches/extrapolation_caches.jl

```diff
@@ -47,3 +47,258 @@ function alg_cache(alg::RichardsonEuler,u,rate_prototype,uEltypeNoUnits,uBottomE
+          extrapolation_weights_2[1:m-1, m-1] = -extrapolation_weights_2[1:m-1, m-1] .// nodes[2:m]
+      end
+  end
+
+  # Compute scaling factors for internal extrapolation operators
+  extrapolation_scalars_2 = ones(Rational{BigInt}, n_max)
+  extrapolation_scalars_2[1] = -nodes[2]
+  for n = 1:(n_max-1)
+      extrapolation_scalars_2[n+1] = -extrapolation_scalars_2[n] * nodes[n+2]
+  end
+
+  # Compute scaling factors for extrapolation operators
+  extrapolation_scalars = -nodes[1] * [BigInt(1); extrapolation_scalars_2]
+
+  # Initialize the constant cache
+  ExtrapolationMidpointDeuflhardConstantCache(Q, n_curr, n_old, subdividing_sequence, stage_number, extrapolation_weights, extrapolation_scalars, extrapolation_weights_2, extrapolation_scalars_2)
+end
+
+@cache mutable struct ExtrapolationMidpointDeuflhardCache{uType,uNoUnitsType,rateType,QType} <: OrdinaryDiffEqMutableCache
+  # Values that are mutated
+  u_tilde::uType
+  u_temp1::uType
+  u_temp2::uType
+  tmp::uType # for get_tmp_cache()
+  T::Array{uType,1}  # Storage for the internal discretisations obtained by the explicit midpoint rule
+  res::uNoUnitsType # Storage for the scaled residual of u and u_tilde
+
+  fsalfirst::rateType
+  k::rateType
+
+  # Begin of constant cache
+  Q::Vector{QType} # Storage for stepsize scaling factors. Q[n] contains information for extrapolation order (n + alg.n_min - 1)
+  n_curr::Int64 # Storage for the current extrapolation order
+  n_old::Int64 # Storage for the extrapolation order n_curr before perfom_step! changes the latter
+
+  # Constant values
+  subdividing_sequence::Array{BigInt,1}  # subdividing_sequence[n] is used for the (n -1)th internal discretisation
+  stage_number::Array{Int,1} # Stage_number[n] contains information for extrapolation order (n + alg.n_min - 1)
+  # Weights and Scaling factors for extrapolation operators
+  extrapolation_weights::Array{Rational{BigInt},2}
+  extrapolation_scalars::Array{Rational{BigInt},1}
+  # Weights and scaling factors for internal extrapolation operators (used for error estimate)
+  extrapolation_weights_2::Array{Rational{BigInt},2}
+  extrapolation_scalars_2::Array{Rational{BigInt},1}
+end
+
+function alg_cache(alg::ExtrapolationMidpointDeuflhard,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,::Type{Val{true}})
+  u_tilde = zero(u)
+  u_temp1 = zero(u)
+  u_temp2 = zero(u)
+  tmp = zero(u)
+  T = fill(zero(u), alg.n_max + 1)
+  res = uEltypeNoUnits.(zero(u))
+  fsalfirst = zero(rate_prototype)
+  k = zero(rate_prototype)
+
+  cc = alg_cache(alg,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,Val{false})
+
+  ExtrapolationMidpointDeuflhardCache(u_tilde, u_temp1, u_temp2, tmp, T, res, fsalfirst, k,
+      cc.Q, cc.n_curr, cc.n_old, cc.subdividing_sequence, cc.stage_number, cc.extrapolation_weights, cc.extrapolation_scalars, cc.extrapolation_weights_2, cc.extrapolation_scalars_2)
```

Comment:
it might be easier to just stick the `cc` in here, using composition to improve code re-use. But again with the other comment, optimizing code simplicity and re-use can come in a later PR.

Reply:
Here I decided against that to make the call `@unpack subdividing_sequence = cache` and such possible for both caches.

Would you prefer an `if - else` statement before calling `@unpack` in order to check if a cache is mutable or not?